devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/ImagePush.java

```diff
@@ -31,6 +32,9 @@ public List<Dependency> forcedDependencies() {
 
     @Override
     public Map<String, String> forcedProperties() {
+        if (dryRun) {
+            return Collections.emptyMap();
+        }
```

Comment:
Similar here. Dry run should allow you to inspect behavior.. you'd want to be able to debug or verify what the forced property list is.

Picture future you with a user that this isn't working for..